core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -315,6 +315,41 @@ private[spark] object Utils extends Logging {
     dir.getCanonicalFile
   }
 
+  /**
+   * Create a directory that is writable by the group.
+   * Grant the customized permission so the shuffle server can
+   * create subdirs/files within the merge folder.
```

Comment:
I think this comment is not appropriate for a util function. Shall we move this to the caller side in `DiskBlockManager`?

Reply:
@Ngone51 @zhouyejoe How about moving this function to DiskBlockManager completely? For more restrictive permissions than `750` we can  use java api. Here, we are trying to run the command `mkdir` just because we want to create mergeDirectory with `770` permission.

Reply:
I'm fine with it.

Reply:
Moved to DiskBlockManager. Also need to change the access modifiers for MAX_DIR_CREATION_ATTEMPTS in Utils.scala.